messaging/src/main/java/org/axonframework/eventhandling/annotation/AnnotatedEventHandlingComponent.java

```diff
@@ -0,0 +1,137 @@
+     *                                 methods.
+     * @param parameterResolverFactory The strategy for resolving handler method parameter values.
+     */
+    public AnnotatedEventHandlingComponent(@Nonnull T annotatedEventHandler,
+                                           @Nonnull ParameterResolverFactory parameterResolverFactory) {
+        this(annotatedEventHandler,
+             parameterResolverFactory,
+             ClasspathHandlerDefinition.forClass(annotatedEventHandler.getClass()));
+    }
+
+    /**
+     * Wraps the given {@code annotatedEventHandler}, allowing it to be subscribed to a
+     * {@link org.axonframework.eventhandling.EventSink} as a {@link EventHandlingComponent}.
+     *
+     * @param annotatedEventHandler    The object containing the
+     *                                 {@link org.axonframework.eventhandling.annotation.EventHandler} annotated
+     *                                 methods.
+     * @param parameterResolverFactory The strategy for resolving handler method parameter values.
+     * @param handlerDefinition        The handler definition used to create concrete handlers.
+     */
+    @SuppressWarnings("unchecked")
+    public AnnotatedEventHandlingComponent(@Nonnull T annotatedEventHandler,
+                                           @Nonnull ParameterResolverFactory parameterResolverFactory,
+                                           @Nonnull HandlerDefinition handlerDefinition) {
+        this.target = requireNonNull(annotatedEventHandler, "The Annotated Event Handler may not be null");
+        this.model = AnnotatedHandlerInspector.inspectType((Class<T>) annotatedEventHandler.getClass(),
+                                                           parameterResolverFactory,
+                                                           handlerDefinition);
+    }
+
+    @Override
+    public EventHandlerRegistry subscribe(@Nonnull QualifiedName name, @Nonnull EventHandler eventHandler) {
+        throw new UnsupportedOperationException(
+                "This Command Handling Component does not support direct command handler registration."
+        );
+    }
+
+    @Nonnull
+    @Override
+    public MessageStream.Empty<Message<Void>> handle(@Nonnull EventMessage<?> event,
+                                                     @Nonnull ProcessingContext context) {
+        requireNonNull(event, "Event Message may not be null");
+        requireNonNull(context, "Processing Context may not be null");
+        var listenerType = target.getClass();
+        var handler = model.getHandlers(listenerType)
+                           .filter(h -> h.canHandle(event, context))
+                           .findFirst();
+        if (handler.isPresent()) {
+            var interceptor = model.chainedInterceptor(listenerType);
+            var result = interceptor.handle(event, context, target, handler.get());
+            return result.ignored();
```

Comment:
After working on that I'm not sure if it's OK that we enforce `MessageStream.Empty` as a result type. 
Life will be easier and more extandable if we allow Single here. Empty extends Single, so we can return empty still as we want. But many times when I was developing the Event-Driven apps I'd like to publish a command in reaction to event. I can imagine if we allow Single, I'd be able to add an interceptor which publish every command returned from event handler. Now I cannot, I'm not sure if it's beneficial that we limit it to just Empty.

Reply:
Nice you like it! :) 
Whereas for me it's still little weird, because there are some messages (side effects are invoked), I just ignore the results.

Reply:
Do you think we can name this method betterl? Like `withIngoredResults`? To be more explicit.